tests/sync/session.cpp

```diff
@@ -382,6 +395,119 @@ TEST_CASE("sync: error handling", "[sync]") {
     }
 }
 
+TEST_CASE("sync: stop policy behavior", "[sync]") {
+    using ProtocolError = realm::sync::ProtocolError;
+    if (!EventLoop::has_implementation())
+        return;
+
+    auto cleanup = util::make_scope_exit([=]() noexcept { SyncManager::shared().reset_for_testing(); });
+    // Server is initially stopped so we can control when the session exits the dying state.
+    SyncServer server(false);
+    auto schema = Schema{
+        {"sync_session_object", {
+            {"value 1", PropertyType::Int},
+            {"value 2", PropertyType::Int},
+        }},
+    };
+
+    constexpr int count = 2;
+    std::function<void(Realm::Config&)> add_objects = [=](Realm::Config& config) {
```

Comment:
No need to use `std::function` here. This can just be `auto`.

Why does this use `[=]`? It doesn't look like it needs to capture anything.